haddock-api/src/Haddock/Backends/Xhtml/Decl.hs

```diff
@@ -646,7 +646,7 @@ ppInstanceSigs links splice unicode qual sigs = do
     TypeSig lnames typ <- sigs
     let names = map unLoc lnames
         L loc rtyp = get_type typ
-    return $ ppSimpleSig links splice unicode qual loc names rtyp
+    return $ ppSimpleSig links splice unicode qual (getLoc $ head $ lnames) names rtyp
```

Comment:
I'm not so sure about this. Why not use `loc`, the span associated with the type? It likely makes little difference in the result, but it would be nice to understand why this change is necessary.